docs/howto/rally_benchmarking.md

```diff
@@ -0,0 +1,268 @@
+| description                     | string     | | A description for the scenario.                                                                                                                                       |
+| version                         | string     | | The version of the package to benchmark. If omitted will pick the current version of the package.                                                                     |
+| data_stream.name                | string     | yes | The data stream to benchmark.                                                                                                                                         |
+| warmup_time_period              | duration   |  | Warmup time period. All data prior to this period will be ignored in the benchmark results.                                                                           |
+| corpora.generator.total_events  | uint64     |  | Number of total events to generate. Example: `20000`                                                                                                                  |
+| corpora.generator.template.raw  | string     |  | Raw template for the corpus generator.                                                                                                                                |
+| corpora.generator.template.path | string     |  | Path to the template for the corpus generator. If a `path` is defined, it will override any `raw` template definition.                                                |
+| corpora.generator.template.type | string     |  | Type of the template for the corpus generator. Default `placeholder`.                                                                                                 |
+| corpora.generator.config.raw    | dictionary |  | Raw config for the corpus generator.                                                                                                                                  |
+| corpora.generator.config.path   | string     |  | Path to the config for the corpus generator. If a `path` is defined, it will override any `raw` config definition.                                                    |
+| corpora.generator.fields.raw    | dictionary |  | Raw fields for the corpus generator.                                                                                                                                  |
+| corpora.generator.fields.path   | string     |  | Path to the fields for the corpus generator. If a `path` is defined, it will override any `raw` fields definition.                                                    |
+
+Example:
+
+`logs-benchmark.yml`
+```yaml
+---
+---
+description: Benchmark 20000 events ingested
+data_stream:
+   name: testds
+warmup_time_period: 10s
```

Comment:
As we don't have an elastic-agent shipping data, what data is there that could come in during the warm up period? Do we need this?

Reply:
with my last commit we wait for this period after we setup the benchmark: ie deleting all existing documents in the data stream. then we collect the stats before running rally, refreshing the index. so yes, I'm ok to remove as well since it make it simpler and lower latency